lib/config/src/app/http.rs

```diff
@@ -0,0 +1,31 @@
+/// Defines an HTTP request.
+#[derive(
+    schemars::JsonSchema, serde::Serialize, serde::Deserialize, PartialEq, Eq, Clone, Debug,
+)]
+pub struct HttpRequest {
+    /// Path to the health check endpoint.
+    pub path: String,
+    /// HTTP method to use for the health check.
+    ///
+    /// Defaults to GET.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub method: Option<String>,
```

Comment:
This should be an enum of the following: GET, POST, PUT, HEAD, OPTIONS

Reply:
* It's easier to not have a typing layer here, because we'd need to do conversions into `http` anyway. Keeping this a String was intentional
* HTTP methods are in theory arbitrary, the well-known ones are just the standard

Reply:
> HTTP methods are in theory arbitrary, the well-known ones are just the standard

oooh TIL! This is pretty cool. This took me down a rabbit hole of all the nifty things custom http methods are used for.

Reply:
Closing as I think this is implemented as it should be.
Please re-open if you disagree.